### PR TITLE
Adjustment of the voting scheme and updated viral taxonomy and bugfixes for new release

### PR DESCRIPTION
**Pull request by mberacochea**
Folks. The pipeline seems to be working OK.

Results for the example assembly -> 
[assembly_results.tar.gz](https://github.com/EBI-Metagenomics/emg-viral-pipeline/files/11906373/assembly_results.tar.gz)

I think the review should be focus on the results files, if they look OK we should make a new release.

### CODE REVIEW

**Comment by hoelzer** [comment]
Great, thx @mberacochea ! 

I looked into the results. Looks fine, but I noticed something. In the `08-final/gff/assembly_virify.gff` file we have for `pos.phage.5` the following CDS and ViPhOG hits:

```tsv
pos.phage.5	Prodigal	CDS	1934	3370	.	+	0	ID=pos.phage.5_3;gbkey=CDS;viphog=ViPhOG1981;viphog_taxonomy=Arquatrovirinae
pos.phage.5	Prodigal	CDS	13526	15382	.	+	0	ID=pos.phage.5_19;gbkey=CDS;viphog=ViPhOG9461;viphog_taxonomy=Twortvirinae
pos.phage.5	Prodigal	CDS	15397	18063	.	+	0	ID=pos.phage.5_20;gbkey=CDS;viphog=ViPhOG9903;viphog_taxonomy=Picovirinae
pos.phage.5	Prodigal	CDS	18063	19595	.	+	0	ID=pos.phage.5_21;gbkey=CDS;viphog=ViPhOG9462;viphog_taxonomy=Twortvirinae
pos.phage.5	Prodigal	CDS	22518	23501	.	+	0	ID=pos.phage.5_26;gbkey=CDS;viphog=ViPhOG27646;viphog_taxonomy=Picovirinae
pos.phage.5	Prodigal	CDS	37511	37756	.	+	0	ID=pos.phage.5_54;gbkey=CDS;viphog=ViPhOG9436;viphog_taxonomy=Klosneuvirinae
pos.phage.5	Prodigal	CDS	37765	38172	.	+	0	ID=pos.phage.5_55;gbkey=CDS;viphog=ViPhOG1970;viphog_taxonomy=Sepvirinae
pos.phage.5	Prodigal	CDS	40637	40816	.	+	0	ID=pos.phage.5_62;gbkey=CDS;viphog=ViPhOG4695;viphog_taxonomy=Twortvirinae
```

but in the final taxonomy assignment in `08-final/taxonomy/high_confidence_viral_contigs_prodigal_annotation_taxonomy.tsv` we get

```tsv
contig_ID	genus	subfamily	family	order	class
pos.phage.5			1	Imitervirales	Megaviricetes
```

For me, it's not clear why we get based on these ViPhOGs Imitervirales as annotation, @guille0387 ? 

Am I missing smt?

**Comment by guille0387** [comment]
hey @hoelzer... mmmm that is really weird, there are no ViPhOG hits to any taxa within the Imitervirales and still reported it as that? I have no idea what happened there. I'll try to run the test in my laptop and check what I get.

**Comment by guille0387** [comment]
Oh no, actually Klosneuvirinae is a subfamily within Imitervirales... but it doesn't make sense that it classifies it with only one hit... this is definitely a bug :(

**Comment by mberacochea** [comment]
Ah, well spotted there. I ran it from scratch, so it's using the latest version of the dbs.

**Comment by mberacochea** [comment]
> > Ah, well spotted there. I ran it from scratch, so it's using the latest version of the dbs.
> 
> Just to be sure, what's your nextflow run command? You are using the latest NCBI taxonomy dump file?

I'm using the dev branch and in a fresh dbs folder:
```
nextflow run EBI-Metagenomics/emg-viral-pipeline \
-work-dir [path] 
-r dev 
--databases [path]/dbs \
--fasta "/homes/$USER/.nextflow/assets/EBI-Metagenomics/emg-viral-pipeline/nextflow/test/assembly.fasta" \
--cores 4 -profile ebi
```
it ran the db download process... so it should be using the latest one.

The ncbi download process:
```
$ ls [path]/workdir/5c/2646557b1a2b3aaee97203c052a722/
2022-11-01_ete3_ncbi_tax.sqlite  ete3_ncbi_tax.sqlite
```

**Comment by hoelzer** [comment]
ok thx, looks good as far as I can tell!

**Comment by mberacochea** [comment]
> Hey guys @mberacochea @guille0387, to keep momentum, what's the status here? I think we are almost good and what is just needed is the update of the viphog metadata to v4 as the new default, which @mberacochea already implemented here:
> 
> https://github.com/EBI-Metagenomics/emg-viral-pipeline/pull/106/files
> 
> Then, we could do a merge and a new release (v2.0 ?) also to update the manuscript.
> 
> The other discrepancy I found with the _Imitervirales_ (where I think @guille0387 already had an idea) we could then solve directly after w/ some hot fix.

That works for me. I'm running the test with `assembly.fasta` again just in case, and if it looks good we can merge and release v2.0.0.

**Comment by mberacochea** [comment]
The results -> [assembly_results.tar.gz](https://github.com/EBI-Metagenomics/emg-viral-pipeline/files/12021742/assembly_results.tar.gz)

**Comment by mberacochea** [comment]
OK. The results -> 
[assembly_results.tar.gz](https://github.com/EBI-Metagenomics/emg-viral-pipeline/files/12028133/assembly_results.tar.gz)

Using:
```
- Workdir location:
Jul-12 10:35:22.931 [main] INFO  nextflow.script.BaseScript -   /hps/nobackup/rdf/metagenomics/service-team/users/mbc/virify-testing/workdir
Jul-12 10:35:22.931 [main] INFO  nextflow.script.BaseScript - Databases location:
Jul-12 10:35:22.932 [main] INFO  nextflow.script.BaseScript -   /hps/nobackup/rdf/metagenomics/service-team/users/mbc/virify-testing/dbs[0m
Jul-12 10:35:22.932 [main] INFO  nextflow.script.BaseScript -  
Jul-12 10:35:22.933 [main] INFO  nextflow.script.BaseScript - [2mDev ViPhOG database: v3[0m
Jul-12 10:35:22.934 [main] INFO  nextflow.script.BaseScript - [2mDev Meta database: v4[0m
Jul-12 10:35:22.934 [main] INFO  nextflow.script.BaseScript -  
Jul-12 10:35:22.935 [main] INFO  nextflow.script.BaseScript - [2mOnly run annotation: false[0m
Jul-12 10:35:22.935 [main] INFO  nextflow.script.BaseScript -  
Jul-12 10:35:22.937 [main] INFO  nextflow.script.BaseScript - 
[0;33mWarning: --meta_version v4 does not include the following discontinued virus taxa 
(according to ICTV) anymore and they have been excluded from the dataset.[0m
- Allolevivirus
- Autographivirinae
- Buttersvirus
- Caudovirales
- Chungbukvirus
- Incheonvirus
- Leviviridae
- Levivirus
- Mandarivirus
- Pbi1virus
- Phicbkvirus
- Radnorvirus
- Sitaravirus
- Vidavervirus
- Myoviridae
- Siphoviridae
- Podoviridae
- Viunavirus
- Orthohepevirus
- Klosneuvirus
- Hendrixvirus
- Rubulavirus
- Avulavirus
- Catovirus
- Nucleorhabdovirus
- Viunavirus
- Gammalipothrixvirus
- Peduovirinae
- Sedoreovirinae```

**Comment by hoelzer** [comment]
Hey @mberacochea for me this looks all good! 

I also ran some tests w/ our Neto and Kleiner mocks and also looked fine. 

I opened a new issue regarding the problem we discovered in the context of this PR: https://github.com/EBI-Metagenomics/emg-viral-pipeline/issues/110

And also some problem with the Sankey visual: https://github.com/EBI-Metagenomics/emg-viral-pipeline/issues/109

But both we can solve after the merge of this PR and having a new release `v2.0.0`. 

I pushed some update to the README to this PR to solve https://github.com/EBI-Metagenomics/emg-viral-pipeline/issues/105

**Comment by hoelzer** [comment]
Hey @mberacochea , @guille0387 pushed some further change that was missing from this branch and now the results are consistent. 

We should merge this if all are fine, make a new major release, and then can work on further details via additional fixes.

File: bin/ratio_evalue_table.py
Comment (mberacochea) [inline bin/ratio_evalue_table.py:109]:
```suggestion
                        help="TSV file: additional_data_vpHMMs_v{1,2,3,4}.tsv", required=True)
```

**Comment by mberacochea** [comment]
> Hey @mberacochea , @guille0387 pushed some further change that was missing from this branch and now the results are consistent.
> 
> We should merge this if all are fine, make a new major release, and then can work on further details via additional fixes.

Alright. The changes look good to me :partying_face:. Since I created the PR, I need one of you to approve the PR before merging this.

**Review by hoelzer — APPROVED** [review]
Ah true, approved ;)

**Pull request merged**
